docs/kerberos.md

```diff
@@ -0,0 +1,124 @@
+---
+post_title: Kerberos
+nav_title: Kerberos
+menu_order: 120
+enterprise: 'no'
+---
+
+
+# HDFS Kerberos
+
+Kerberos is an authentication system to allow Spark to retrieve and write data securely to a Kerberos-enabled HDFS cluster. As of Mesosphere Spark `2.2.0-2`, long-running jobs will renew their delegation tokens (authentication credentials). This section assumes you have previously set up a Kerberos-enabled HDFS cluster. For instruction on how to setup a secure HDFS cluster see the [HDFS documentation][http://example.com]
+
+## Spark Installation
+
+Spark (and all Kerberos-enabed) components need a valid `krb5.conf` file. You can setup the Spark service to use a single `krb5.conf` file for all of the it's drivers.
+
+1.  A `krb5.conf` file tells Spark how to connect to your KDC.  Base64 encode this file:
+
+        cat krb5.conf | base64 -w 0 
+
+1.  Put the encoded file (as a string) into your JSON configuration file:
+
+        {
+           "security": {
+             "kerberos": {
+              "krb5conf": "<base64 encoding>"
+              }
+           }
+        }
+        
+     Your configuration will probably also have the `hdfs` parameters from above:
+     
+        {
+          "service": {
+              "name": "kerberized-spark",
+              "user": "nobody"
+          },
+          "hdfs": {
+              "config-url": "http://api.hdfs.marathon.l4lb.thisdcos.directory/v1/endpoints"
+          },
+          "security": {
+              "kerberos": {
+                  "krb5conf": "<base64_encoding>"
+            }
+          }
+        }
+
+        
+1.  Install Spark with your custom configuration, here called `options.json`:
+
+        dcos package install --options=/path/to/options.json spark
+        
+1. If you've enabled the history server via `history-server.enabled`, you must also configure the principal and keytab for the history server.  **WARNING**: The keytab contains secrets, in the current history server package the keytab is not stored securely. See [Limitations][9]
+
+    Base64 encode your keytab:
+
+        cat spark.keytab | base64
+
+    And add the following to your configuration file:
+
+         {
+            "history-server": {
+                "kerberos": {
+                  "principal": "spark@REALM",
+                  "keytab": "<base64 encoding>"
+                }
+            }
+         }
+
```

Comment:
Are we missing a step of adding a keytab or tgt to the secret store?

Reply:
I didn't change any of the instructions w.r.t. the history server because I know that's still a WIP. But I'll add a note about uploading the keytab.